src/Database/Redis/Cluster.hs

```diff
@@ -263,17 +263,30 @@ evaluatePipeline refreshShardmapAction conn@(Connection shardNodeVar infoMap _)
         -- otherwise if there is any response in the connection buffer it'll get forwarded to other requests that are reusing the same connection.
```

Comment:
take a random connection where there are no exceptions.
PERF_CONCERN: Since usually we send only one request at time, this won't be

Shall we udpate this comment block with the new logic we are now using ?